spring-xd-dirt/src/main/java/org/springframework/xd/dirt/stream/zookeeper/ZooKeeperStreamRepository.java

```diff
@@ -282,7 +282,7 @@ public void deleteAll() {
 
 		List<Stream> results = new ArrayList<Stream>();
 		for (Stream stream : all) {
-			if (stream.getDefinition().getName().compareTo(to) > 1) {
+			if (stream.getDefinition().getName().compareTo(to) > 0) {
```

Comment:
So this is the only relevant changed line, and indeed the previous code was broken.
One point still, although it is not the subject of this PR: The `fromInclusive` and `toInclusive` parameters are not honored